### DISCUSSION TITLE
Catch-All Splat route that includes index page

### DISCUSSION THREAD

**Discussion by noahflk** [TanStack/router#1213] [Q&A]
I am familiar with this type of file-based route from Next.js: `pages/sign-up/[[...index]].ts`. It matches both `/sign-up` and `/sign-up/anything`.

The only way I've found to replicate this in TanStack Router is having two separate files: `/sign-up/$.tsx` and `/sign-up/index.tsx`. Is there a way to do it with a single file based route?

**Comment by js2me** [comment]
any updates ?

**Reply by noahflk** [reply]
Might be worth filing an issue. Discussions aren't helping

**Reply by mhuggins** [reply]
+1, I'm running into an issue with this as well.  I need the index page and splat page to share the same react tree, and I'm running into trouble because it seems they are not able to currently.

**Comment by mhuggins** [comment]
I'm trying to redirect from my index page to my splat page, but that seems to just not do anything.  I get stuck on my app's loader, and neither of the page's `beforeLoad` functions get called after the initial `redirect`.

```ts
import { Route as SplatRoute } from "./foo.$";

export const Route = createFileRoute("/foo/")({
  beforeLoad: async () => {
    throw redirect({
      to: SplatRoute.fullPath,
      params: { _splat: "" },
    });
  },
});
```
